modules/geo/src/main/java/org/opensearch/geo/algorithm/PolygonGenerator.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.algorithm;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.util.CollectionUtils;
+
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.IntStream;
+
+/**
+ * Helper class to generate a polygon. Keeping this in the src folder so that GeoSpatial plugin can take advantage of
+ * this helper to create the Polygons, rather than hardcoding the values.
+ */
+public class PolygonGenerator {
```

Comment:
Is this really just a utility class to generate a convex hull from a collection of points? Should we add coordinate safety and only accept a list of `Point` and `XYPoint` cooridnates and generate the corresponding `Polygon` or `XYPolygon` based on those input coordinates? This would add convex hull validation that's not captured here.

Reply:
1. No it's not that. Within convex hull, polygon(convex hull) can contain vertices which is less than the total number of vertices. But in this algo the number of vertices remain same. Please check the documentation on the class ValtrAlgorithm present in this same file. It has some visual links how this algo generates the ploygon. Adding the same link here for reference : https://observablehq.com/@tarte0/generate-random-simple-polygon. 

2. This call was mainly created for the tests, hence I feel we can keep it like this where it accepts a bunch of list in double array. Moreover we will be adding the classes in Tests which will provide the abstraction on top of this class. I exposed this in src directory so that other can take advantage of this class and build their own validations on top of it as it is possible that other Point classes are not part of core. Example: `XYPoint` class is present in GeoSpatial plugin.

Reply:
> This call was mainly created for the tests, hence I feel we can keep it like this where it accepts a bunch of list in double array. Moreover we will be adding the classes in Tests which will provide the abstraction on top of this class.

:+1:  makes sense. I don't have a problem iterating in follow up PRs